src/features/weak-references.md

```diff
@@ -217,8 +217,6 @@ After hearing about these new capabilities, it might be tempting to `WeakRef` Al
 
 In general, avoid writing code that depends on the garbage collector cleaning up a `WeakRef` or calling a finalizer at any predictable time — [it can’t be done](https://github.com/tc39/proposal-weakrefs#a-note-of-caution)! Moreover, whether an object is garbage-collectible at all may depend on implementation details, such as the representation of closures, that are both subtle and may differ across JavaScript engines and even between different versions of the same engine.
 
-For example, don’t place important logic in the code path of a finalizer. There’s no way to predict _when_, or even _if_, a given finalizer gets called. It’s best to think of `WeakRef`s and finalizers as **progressive enhancement**: it’s nice if your custom finalizer code runs, but your program should still work without it.
```

Comment:
I think it's good to keep something like this, but say something closer to "it's not guaranteed to run if e.g. you close your browser window", and that it should be used only for cleanup and not for business logic (e.g. a dependable signal POSTed back to the server that some object was collected, as you allude to with "metrics" above).

I can't quite think of a way to phrase it, but basically the dev has to make sure that the they don't rely on the finalizer more than on the GC.

Reply:
I've tried to address this